Analysis/emission_line_fit.py

```diff
@@ -335,20 +313,18 @@ def zm_general(fitspath, Spect_1D, dispersion, wave, lambda0, line_type, line_na
     pdf_pages = PdfPages(outpdf)
     table0 = asc.read(fitspath + filename_dict['bin_info'], format='fixed_width_two_line')
     out_ascii = fitspath + filename_dict['bin_fit']
-    ID = table0[bin_names0[0]]
+    ID = table0[bin_names0[0]].data
     N = table0[bin_names0[1]].data
     
+    em_dict = OrderedDict([(bin_names0[0], ID)])
     for ii in range(len(lambda0)):
         curr_line_cols = [line_col for line_col in gauss_lines_names0 if line_col.startswith(line_name[ii])]
-        em_table = zoom_gauss_plot(pdf_pages, N, wave, Spect_1D, dispersion, s2, lambda0, lambda0[ii], 
-                                   curr_line_cols, line_type=line_type[ii], line_name=line_name[ii], 
+        line_dict = create_empty_dict(curr_line_cols, arr_size=len(N), ordered=True)
```

Comment:
You might consider having `line_dict` be defined in `zoom_gauss_plot` rather than passing that in.